src/commands/documents.ts

```diff
@@ -64,29 +84,71 @@ const createDocuments = (n: number, generated: number, createDoc: DocumentCreato
 };
 
 
-export const generateAlerts = async (n: number) => {
-  await indexCheck(ALERT_INDEX, alertMappings as MappingTypeMapping);
+export const generateAlerts = async (alertCount: number, hostCount: number, userCount: number) => {
+
+  if (userCount > alertCount) {
+    console.log('User count should be less than alert count');
+    process.exit(1);
+  }
 
-  console.log('Generating alerts...');
+  if (hostCount > alertCount) {
+    console.log('Host count should be less than alert count');
+    process.exit(1);
+  }
 
-  await generateDocs({
-    createDocs: createAlerts,
-    amount: n,
-    index: ALERT_INDEX,
-  });
+  console.log(`Generating ${alertCount} alerts containing ${hostCount} hosts and ${userCount} users.`);
+  const concurrency = 10; // how many batches to send in parallel
+  const batchSize = 2500; // number of alerts in a batch
+  const no_overrides = {};
+
+  const batchOpForIndex = ({ userName, hostName } : { userName: string, hostName: string }) => alertToBatchOps(createAlerts(no_overrides, { userName, hostName }), ALERT_INDEX);
+
+
+  console.log('Generating entity names...');
+  const userNames = Array.from({ length: userCount }, () => faker.internet.userName());
+  const hostNames = Array.from({ length: hostCount }, () => faker.internet.domainName());
+
+  console.log('Assigning entity names...')
+  const alertEntityNames = Array.from({ length: alertCount }, (_, i) => ({
+    userName: userNames[i % userCount],
+    hostName: hostNames[i % hostCount],
+  }));
+  
+  console.log('Entity names assigned. Batching...');
+  const operationBatches = _.chunk(alertEntityNames, batchSize).map((batch) => 
+    batch.flatMap(batchOpForIndex)
+  );
```

Comment:
I wish we didn't have to do all this pre-processing beforehand. This code consumes a lot of memory and processing time when generating millions of alerts. I wonder if doing all this pre-processing to generate pretty user and host names is worth it.

Reply:
I do prefer the entity names but I know there is no real advantage, if this wasn't a dev script I'd be more open to memory/cpu concerns, it takes 70s to generate 1 million alerts which I think is fine?